absolute_paths_in_stderr does not rewrite `    inlined from` messages

### How to reproduce ###

I'll call this file `test.cpp`:
```cpp
#include <array>

int foo(std::array<int, 8> const & arr)
{
    return arr[9];
}

int run(std::array<int, 8> const & arr)
{
    return foo(arr);
}

int main()
{
    std::array<int, 8> arr{};
    return run(arr);
}
```

I'm using `/tmp/ccache` as an example directory. The cpp file is then in `/tmp/ccache/test.cpp`.
Additionally, I create `/tmp/ccache/sub` such that relative paths are created in the examples.

**The important part is lines in the form of `    inlined from <signature> at test.cpp`.**

## Running from `/tmp/ccache`

### gcc
```
# /tmp/ccache$ g++-15 -Wall -O2 -c test.cpp -o test.cpp.o
In function ‘int foo(const std::array<int, 8>&)’,
    inlined from ‘int run(const std::array<int, 8>&)’ at test.cpp:10:15,
    inlined from ‘int main()’ at test.cpp:16:15:
test.cpp:5:17: warning: array subscript 9 is outside array bounds of ‘std::array<int, 8> [1]’ [-Warray-bounds=]
    5 |     return arr[9];
      |                 ^
test.cpp: In function ‘int main()’:
test.cpp:15:24: note: at offset 36 into object ‘arr’ of size 32
   15 |     std::array<int, 8> arr{};
      |                        ^~~
```

### ccache
```
# /tmp/ccache$ ccache g++-15 -Wall -O2 -c test.cpp -o test.cpp.o
In function ‘int foo(const std::array<int, 8>&)’,
    inlined from ‘int run(const std::array<int, 8>&)’ at test.cpp:10:15,
    inlined from ‘int main()’ at test.cpp:16:15:
/tmp/ccache/test.cpp:5:17: warning: array subscript 9 is outside array bounds of ‘std::array<int, 8> [1]’ [-Warray-bounds=]
    5 |     return arr[9];
      |                 ^
/tmp/ccache/test.cpp: In function ‘int main()’:
/tmp/ccache/test.cpp:15:24: note: at offset 36 into object ‘arr’ of size 32
   15 |     std::array<int, 8> arr{};
      |                        ^~~
```

### expected
```
# /tmp/ccache$ ccache g++-15 -Wall -O2 -c test.cpp -o test.cpp.o
In function ‘int foo(const std::array<int, 8>&)’,
    inlined from ‘int run(const std::array<int, 8>&)’ at /tmp/ccache/test.cpp:10:15,
    inlined from ‘int main()’ at /tmp/ccache/test.cpp:16:15:
/tmp/ccache/test.cpp:5:17: warning: array subscript 9 is outside array bounds of ‘std::array<int, 8> [1]’ [-Warray-bounds=]
    5 |     return arr[9];
      |                 ^
/tmp/ccache/test.cpp: In function ‘int main()’:
/tmp/ccache/test.cpp:15:24: note: at offset 36 into object ‘arr’ of size 32
   15 |     std::array<int, 8> arr{};
      |                        ^~~
```

## Running from `/tmp/ccache/sub`

### gcc
```
# /tmp/ccache/sub$ g++-15 -Wall -O2 -c ../test.cpp -o test.cpp.o
In function ‘int foo(const std::array<int, 8>&)’,
    inlined from ‘int run(const std::array<int, 8>&)’ at ../test.cpp:10:15,
    inlined from ‘int main()’ at ../test.cpp:16:15:
../test.cpp:5:17: warning: array subscript 9 is outside array bounds of ‘std::array<int, 8> [1]’ [-Warray-bounds=]
    5 |     return arr[9];
      |                 ^
../test.cpp: In function ‘int main()’:
../test.cpp:15:24: note: at offset 36 into object ‘arr’ of size 32
   15 |     std::array<int, 8> arr{};
      |                        ^~~
```

### ccache
```
# /tmp/ccache/sub$ ccache g++-15 -Wall -O2 -c ../test.cpp -o test.cpp.o
In function ‘int foo(const std::array<int, 8>&)’,
    inlined from ‘int run(const std::array<int, 8>&)’ at ../test.cpp:10:15,
    inlined from ‘int main()’ at ../test.cpp:16:15:
/tmp/ccache/test.cpp:5:17: warning: array subscript 9 is outside array bounds of ‘std::array<int, 8> [1]’ [-Warray-bounds=]
    5 |     return arr[9];
      |                 ^
/tmp/ccache/test.cpp: In function ‘int main()’:
/tmp/ccache/test.cpp:15:24: note: at offset 36 into object ‘arr’ of size 32
   15 |     std::array<int, 8> arr{};
      |                        ^~~
```

### expected
```
In function ‘int foo(const std::array<int, 8>&)’,
    inlined from ‘int run(const std::array<int, 8>&)’ at /tmp/ccache/test.cpp:10:15,
    inlined from ‘int main()’ at /tmp/ccache/test.cpp:16:15:
/tmp/ccache/test.cpp:5:17: warning: array subscript 9 is outside array bounds of ‘std::array<int, 8> [1]’ [-Warray-bounds=]
    5 |     return arr[9];
      |                 ^
/tmp/ccache/test.cpp: In function ‘int main()’:
/tmp/ccache/test.cpp:15:24: note: at offset 36 into object ‘arr’ of size 32
   15 |     std::array<int, 8> arr{};
      |                        ^~~
```

### Actual behavior ###
The path in `inlined from <signature> at <path>` is not absolute.

### Expected behavior ###
It would be nice if the path were absolute.

### Environment ###

Current master (f8919f6507032a3b9739187d23c31622723e57b1)

`ccache version master.f8919f65`

### Misc ###

Locally, I fixed it like this, though I'm not sure whether this is the best way to do it.

```diff
diff --git a/src/ccache/core/common.cpp b/src/ccache/core/common.cpp
index 0a6ac2b8..0b17f46f 100644
--- a/src/ccache/core/common.cpp
+++ b/src/ccache/core/common.cpp
@@ -105,6 +105,8 @@ rewrite_stderr_to_absolute_paths(std::string_view text)
     "of module ",
     "imported at ",
   };
+  static const std::string_view inlined_from_msg = "    inlined from ";
+  static const std::string_view inlined_from_msg_separator = " at ";
 
   std::string result;
   using util::Tokenizer;
@@ -112,11 +114,20 @@ rewrite_stderr_to_absolute_paths(std::string_view text)
                              "\n",
                              Tokenizer::Mode::include_empty,
                              Tokenizer::IncludeDelimiter::yes)) {
-    for (const auto& in_file_included_from : in_file_included_from_msgs) {
-      if (util::starts_with(line, in_file_included_from)) {
-        result += in_file_included_from;
-        line = line.substr(in_file_included_from.length());
-        break;
+    if (util::starts_with(line, inlined_from_msg)) {
+      size_t signature_end = line.find(inlined_from_msg_separator);
+      if (signature_end != std::string_view::npos) {
+        signature_end += inlined_from_msg_separator.size();
+        result.append(line.data(), signature_end);
+        line = line.substr(signature_end);
+      }
+    } else {
+      for (const auto& in_file_included_from : in_file_included_from_msgs) {
+        if (util::starts_with(line, in_file_included_from)) {
+          result += in_file_included_from;
+          line = line.substr(in_file_included_from.length());
+          break;
+        }
       }
     }
     while (!line.empty() && line[0] == 0x1b) {
diff --git a/unittest/test_core_common.cpp b/unittest/test_core_common.cpp
index 9241cf4b..96441b20 100644
--- a/unittest/test_core_common.cpp
+++ b/unittest/test_core_common.cpp
@@ -80,7 +80,9 @@ TEST_CASE("core::rewrite_stderr_to_absolute_paths")
     "In module imported at \x1b[01m\x1b[Kexisting:\x1b[m\x1b[K: foo\n"
     "In module imported at \x1b[01m\x1b[Kexisting:47:11:\x1b[m\x1b[K: foo\n"
     "imported at \x1b[01m\x1b[Kexisting:\x1b[m\x1b[K: foo\n"
-    "imported at \x1b[01m\x1b[Kexisting:47:11:\x1b[m\x1b[K: foo\n";
+    "imported at \x1b[01m\x1b[Kexisting:47:11:\x1b[m\x1b[K: foo\n"
+    "    inlined from foo at \x1b[01m\x1b[Kexisting:\x1b[m\x1b[K\n"
+    "    inlined from foo at \x1b[01m\x1b[Kexisting:47:11,\x1b[m\x1b[K\n";
   std::string expected = FMT(
     "a:1:2\n"
     "a(3):\n"
@@ -109,7 +111,9 @@ TEST_CASE("core::rewrite_stderr_to_absolute_paths")
     "In module imported at \x1b[01m\x1b[K{0}:\x1b[m\x1b[K: foo\n"
     "In module imported at \x1b[01m\x1b[K{0}:47:11:\x1b[m\x1b[K: foo\n"
     "imported at \x1b[01m\x1b[K{0}:\x1b[m\x1b[K: foo\n"
-    "imported at \x1b[01m\x1b[K{0}:47:11:\x1b[m\x1b[K: foo\n",
+    "imported at \x1b[01m\x1b[K{0}:47:11:\x1b[m\x1b[K: foo\n"
+    "    inlined from foo at \x1b[01m\x1b[K{0}:\x1b[m\x1b[K\n"
+    "    inlined from foo at \x1b[01m\x1b[K{0}:47:11,\x1b[m\x1b[K\n",
     *fs::canonical("existing"));
   CHECK(core::rewrite_stderr_to_absolute_paths(input) == expected);
 }
```

--- a/src/ccache/core/common.cpp
+++ b/src/ccache/core/common.cpp
@@ -90,8 +90,32 @@ make_relative_path(const Context& ctx, const fs::path& path)
   }
 }
 
-std::string
-rewrite_stderr_to_absolute_paths(std::string_view text)
+inline bool
+parse_inlined_from_msg(std::string_view& line, std::string& result)
+{
+  // Reference for GCC: <https://github.com/gcc-mirror/gcc/blob/
+  // c7507e395f096240ffa8fa5dfcbfcfd8c5e23bb8/gcc/langhooks.cc#L450-L467>
+  static const std::string_view inlined_from_msg = "    inlined from ";
+  static const std::string_view inlined_from_msg_separator = " at ";
+
+  if (!util::starts_with(line, inlined_from_msg)) {
+    return false;
+  }
+
+  size_t signature_end = line.find(inlined_from_msg_separator);
+  if (signature_end == std::string_view::npos) {
+    return false;
+  }
+
+  signature_end += inlined_from_msg_separator.size();
+  result.append(line.data(), signature_end);
+  line = line.substr(signature_end);
+
+  return true;
+}
+
+inline bool
+parse_in_file_included_from_msg(std::string_view& line, std::string& result)
 {
   // Line prefixes from GCC plus extra space at the end. Reference:
   // <https://gcc.gnu.org/git?p=gcc.git;a=blob;f=gcc/diagnostic-format-text.cc;
@@ -106,18 +130,28 @@ rewrite_stderr_to_absolute_paths(std::string_view text)
     "imported at ",
   };
 
+  for (const auto& in_file_included_from : in_file_included_from_msgs) {
+    if (util::starts_with(line, in_file_included_from)) {
+      result += in_file_included_from;
+      line = line.substr(in_file_included_from.length());
+      return true;
+    }
+  }
+
+  return false;
+}
+
+std::string
+rewrite_stderr_to_absolute_paths(std::string_view text)
+{
   std::string result;
   using util::Tokenizer;
   for (auto line : Tokenizer(text,
                              "\n",
                              Tokenizer::Mode::include_empty,
                              Tokenizer::IncludeDelimiter::yes)) {
-    for (const auto& in_file_included_from : in_file_included_from_msgs) {
-      if (util::starts_with(line, in_file_included_from)) {
-        result += in_file_included_from;
-        line = line.substr(in_file_included_from.length());
-        break;
-      }
+    if (!parse_inlined_from_msg(line, result)) {
+      parse_in_file_included_from_msg(line, result);
     }
     while (!line.empty() && line[0] == 0x1b) {
       auto csi_seq = find_first_ansi_csi_seq(line);

--- a/unittest/test_core_common.cpp
+++ b/unittest/test_core_common.cpp
@@ -80,7 +80,9 @@ TEST_CASE("core::rewrite_stderr_to_absolute_paths")
     "In module imported at \x1b[01m\x1b[Kexisting:\x1b[m\x1b[K: foo\n"
     "In module imported at \x1b[01m\x1b[Kexisting:47:11:\x1b[m\x1b[K: foo\n"
     "imported at \x1b[01m\x1b[Kexisting:\x1b[m\x1b[K: foo\n"
-    "imported at \x1b[01m\x1b[Kexisting:47:11:\x1b[m\x1b[K: foo\n";
+    "imported at \x1b[01m\x1b[Kexisting:47:11:\x1b[m\x1b[K: foo\n"
+    "    inlined from foo at \x1b[01m\x1b[Kexisting:\x1b[m\x1b[K\n"
+    "    inlined from foo at \x1b[01m\x1b[Kexisting:47:11,\x1b[m\x1b[K\n";
   std::string expected = FMT(
     "a:1:2\n"
     "a(3):\n"
@@ -109,7 +111,9 @@ TEST_CASE("core::rewrite_stderr_to_absolute_paths")
     "In module imported at \x1b[01m\x1b[K{0}:\x1b[m\x1b[K: foo\n"
     "In module imported at \x1b[01m\x1b[K{0}:47:11:\x1b[m\x1b[K: foo\n"
     "imported at \x1b[01m\x1b[K{0}:\x1b[m\x1b[K: foo\n"
-    "imported at \x1b[01m\x1b[K{0}:47:11:\x1b[m\x1b[K: foo\n",
+    "imported at \x1b[01m\x1b[K{0}:47:11:\x1b[m\x1b[K: foo\n"
+    "    inlined from foo at \x1b[01m\x1b[K{0}:\x1b[m\x1b[K\n"
+    "    inlined from foo at \x1b[01m\x1b[K{0}:47:11,\x1b[m\x1b[K\n",
     *fs::canonical("existing"));
   CHECK(core::rewrite_stderr_to_absolute_paths(input) == expected);
 }

